src/Microsoft.OData.Edm/Csdl/CsdlJsonWriter.cs

```diff
@@ -0,0 +1,209 @@
+                    // End of each reference document.
+                    this.jsonWriter.WriteEndObject();
+                }
+
+                // End of $Reference
+                this.jsonWriter.WriteEndObject();
+            }
+        }
+
+        /// <summary>
+        /// $Include Array.
+        /// </summary>
+        private void WriteIncludes(IEnumerable<IEdmInclude> includes)
+        {
+            if (includes == null || !includes.Any())
+            {
+                return;
+            }
+
+            // The reference object MAY contain the members $Include
+            this.jsonWriter.WritePropertyName("$Include");
+
+            // The value of $Include is an array.
+            this.jsonWriter.WriteStartArray();
+
+            foreach (IEdmInclude include in includes)
+            {
+                // Array items are objects.
+                this.jsonWriter.WriteStartObject();
+
+                // MUST contain the member $Namespace
+                this.jsonWriter.WriteRequiredProperty("$Namespace", include.Namespace);
+
+                // MAY contain the member $Alias
+                this.jsonWriter.WriteOptionalProperty("$Alias", include.Alias);
+
+                // The include object MAY contain annotations.
+                // So far, it's not supported in ODL.
+
+                this.jsonWriter.WriteEndObject();
+            }
+
+            this.jsonWriter.WriteEndArray();
+        }
+
+        /// <summary>
+        /// $IncludeAnnotations Array.
+        /// </summary>
+        private void WriteIncludeAnnotations(IEnumerable<IEdmIncludeAnnotations> includeAnnotations)
+        {
+            if (includeAnnotations == null || !includeAnnotations.Any())
+            {
+                return;
+            }
+
+            // The reference object MAY contain the members $IncludeAnnotations
+            this.jsonWriter.WritePropertyName("$IncludeAnnotations");
+
+            // The value of $IncludeAnnotations is an array.
+            this.jsonWriter.WriteStartArray();
+
+            foreach (IEdmIncludeAnnotations includeAnnotation in includeAnnotations)
+            {
+                // Array items are objects
+                this.jsonWriter.WriteStartObject();
+
+                // MUST contain the member $TermNamespace
+                this.jsonWriter.WriteRequiredProperty("$TermNamespace", includeAnnotation.TermNamespace);
+
+                // MAY contain the members $Qualifier
+                this.jsonWriter.WriteOptionalProperty("$Qualifier", includeAnnotation.Qualifier);
+
+                // MAY contain the members $TargetNamespace
+                this.jsonWriter.WriteOptionalProperty("$TargetNamespace", includeAnnotation.TargetNamespace);
+
+                this.jsonWriter.WriteEndObject();
+            }
+
+            this.jsonWriter.WriteEndArray();
+        }
+
+        /// <summary>
+        /// Schema Object.
+        /// </summary>
+        private void WriteSchemata()

```

Comment:
"Schemata", is this a typo?

Reply:
Emm, No, here means to write multiple schema (schemata: plural) @mikepizzo ， would you double confirm it?

Reply:
@xuzhg Wow, today I learned: schema (plural schemata or schemas).

Reply:
Me too, english is not my native Language. 

In my memory, i got this name from: 
http://docs.oasis-open.org/odata/odata/v4.0/errata02/os/complete/models/MetadataService.edmx

```xml
<EntityContainer Name="MetadataContainer">
        <EntitySet Name="Schemata" EntityType="Meta.Schema">
          <NavigationPropertyBinding Target="EntityContainer" Path="EntityContainer" />
          <NavigationPropertyBinding Target="Terms" Path="Terms" />
          <NavigationPropertyBinding Target="Actions" Path="Actions" />
          <NavigationPropertyBinding Target="Functions" Path="Functions" />
          <NavigationPropertyBinding Target="Annotations" Path="Annotations" />
          <NavigationPropertyBinding Target="References" Path="Reference" />
          <NavigationPropertyBinding Target="Types" Path="Types" />
        </EntitySet>
```

There's a line mentioned: an entity set named "Schemata", Its type is "Schema".

Reply:
yes; "Schemata" is an appropriate use here

---
In reply to: [439796912](https://github.com/OData/odata.net/pull/1775#discussion_r439796912) [](ancestors = 439796912)

Reply:
:smile: 

---
In reply to: [439797106](https://github.com/OData/odata.net/pull/1775#discussion_r439797106) [](ancestors = 439797106)